test/mobile/features/scripts/generate_xcode_project.sh

```diff
@@ -16,7 +16,21 @@ pushd "$script_path/../fixtures"
 
 # Run unity and immediately exit afterwards, log all output, disable the
 # package manager (we just don't need it and it slows things down)
-DEFAULT_CLI_ARGS="-quit -batchmode -logFile unity.log -noUpm"
+DEFAULT_CLI_ARGS="-quit -batchmode -logFile unity.log"
+
+if [[ "$UNITY_PATH" == *"2020"* ]]; then
+  # Copy the 2020 package manifest in for that version
+  mkdir -p maze_runner/Packages
+  cp manifests/manifest_2020.json maze_runner/Packages/manifest.json
+elif [[ "$UNITY_PATH" == *"2019"* ]]; then
+  # Copy the 2019 package manifest in for that version
+  mkdir -p maze_runner/Packages
+  cp manifests/manifest_2019.json maze_runner/Packages/manifest.json
+else
+  # On all other versions disable the package manager
+  DEFAULT_CLI_ARGS="${DEFAULT_CLI_ARGS} -noUpm"
+fi
```

Comment:
Is there any way to avoid adding duplication between this script and `build_android.sh`/`build_maze_runner.sh`?

Reply:
Yes, but not without a small re-write of them both.  Regarding both of your queries a couple of tickets, one for the multi-stage builds and one for the scripting update would be good if you're happy to create those?